clouddq/classes/dq_rule_binding.py

```diff
@@ -316,12 +336,25 @@ def resolve_all_configs_to_dict(
                     rule_configs_dict[rule_id] = rule_config
             # Resolve filter configs
             row_filter_config = self.resolve_row_filter_config(configs_cache)
+            # resolve reference columns config
+            reference_columns_config = self.resolve_reference_columns_config(
+                configs_cache
+            )
+            include_reference_columns = (
+                reference_columns_config.include_reference_columns.strip("[]")
+                .replace(" ", "")
+                .replace('"', "")
+                .split(",")
```

Comment:
move all string parsing to list logic inside of the class DqReferenceColumns. `reference_columns_config.include_reference_columns` should already return a list, not a string as it is right now.

Reply:
this logic needs to happen right at the beginning where you either parse the user input from YAML or retrieving the data from sqlite